### PR TITLE
addons/git: fix interaction with git's safe.directory

### PR DESCRIPTION
**Pull request by arthurzam**
When using GitLog class, it runs the git commands with clean git config, (aka it uses an empty one) which results in inability to use pkgcheck scan on overlays (as it tries to access system's gentoo repo, which is owned by root). If the user set safe.directory in his config, it will still not work, as we override user's config.

Fix it by creating a temporary file, which we fill with minimal config which disables safe.directory feature, and passing it as the config file to the git log command.

Resolves: https://github.com/pkgcore/pkgcheck/issues/412

### CODE REVIEW

File: src/pkgcheck/addons/git.py [hunk 1]
```diff
@@ -79,14 +76,37 @@ def __init__(self, *args, commit):
         self.commit = commit
 
 
+class GitSafeDirectory:
```
Comment (laumann) [inline src/pkgcheck/addons/git.py:79]: [suggestion] This is meant to be used to override whatever configuration the user has, maybe a better name is `GitGlobalConfig` or something to that effect? Or just `GitConfig`.

**Comment by arthurzam** [comment]
@thesamesam does this solution make sense for you?

File: src/pkgcheck/addons/git.py [hunk 2]
```diff
@@ -79,14 +76,37 @@ def __init__(self, *args, commit):
         self.commit = commit
 
 
+class GitConfig:
+    """Manages temporary file which holds git config for disabling
+    safe directory feature of git."""
+
+    def __init__(self):
+        fd, self.path = tempfile.mkstemp()
+        print('AAA', self.path)
```
Comment (thesamesam) [inline src/pkgcheck/addons/git.py:85]: ?

Comment (laumann) [reply src/pkgcheck/addons/git.py:85]: Looks like debugging :-)

File: src/pkgcheck/addons/git.py [hunk 3]
```diff
@@ -79,14 +76,37 @@ def __init__(self, *args, commit):
...
+    def close(self):
+        os.unlink(self.path)
+
+
 class GitLog:
     """Iterator for decoded `git log` line output."""
 
     def __init__(self, cmd, path):
         self._running = False
+        self.safe_directory = GitConfig()
```
Comment (laumann) [inline src/pkgcheck/addons/git.py:106]: [nit] I would maybe rename this variables to `self.git_config` so it'd be
```python
env=self.git_config.config_env
```
in the call to `subprocens.Popen`

**Review by laumann — COMMENTED** [review]
Looks good, the debugging print line should probably go. Nice work.

File: src/pkgcheck/addons/git.py (hunk 2 above)
Comment (arthurzam) [reply src/pkgcheck/addons/git.py]: Oh, I missed this line.
Yes, this was for debugging the auto delete part.

File: src/pkgcheck/addons/git.py (hunk 3 above)
Comment (arthurzam) [reply src/pkgcheck/addons/git.py]: done

**Review by thesamesam — APPROVED** [review]
We might need to revisit whether we really want to ignore 'git log' in future (probably by revisiting git bindings in future).

Unfortunately, not that long ago at all, when radhermit tried to use the various git Python bindings out there, IIRC they were all far slower and had other limitations. But w/e.

This is what I was expecting and looks good, thanks!

**Pull request merged**